public/scripts/extensions/connection-manager/index.js

```diff
@@ -208,15 +214,36 @@ async function readProfileFromCommands(mode, profile, cleanUp = false) {
 async function createConnectionProfile(forceName = null) {
     const mode = main_api === 'openai' ? 'cc' : 'tc';
     const id = uuidv4();
+    /** @type {ConnectionProfile} */
     const profile = {
         id,
         mode,
+        exclude: [],
     };
 
     await readProfileFromCommands(mode, profile);
 
     const profileForDisplay = makeFancyProfile(profile);
-    const template = await renderExtensionTemplateAsync(MODULE_NAME, 'profile', { profile: profileForDisplay });
+    const template = $(await renderExtensionTemplateAsync(MODULE_NAME, 'profile', { profile: profileForDisplay }));
+    template.find('input[name="exclude"]').on('input', function () {
+        const fancyName = String($(this).val());
+        const keyName = Object.entries(FANCY_NAMES).find(x => x[1] === fancyName)?.[0];
+        if (!keyName) {
+            console.warn('Key not found for fancy name:', fancyName);
+            return;
+        }
+
+        if (!Array.isArray(profile.exclude)) {
+            profile.exclude = [];
+        }
+
+        const excludeState = !$(this).prop('checked');
+        if (excludeState) {
+            profile.exclude.push(keyName);
+        } else {
+            profile.exclude.splice(profile.exclude.indexOf(keyName), 1);
```

Comment:
This will splice off the last element of your array if the keyname is not found.
![image](https://github.com/user-attachments/assets/a11ce602-6b42-466a-9e78-4220e1e8c955)
```suggestion
            const index = profile.exclude.indexOf(keyName);
            index !== -1 && profile.exclude.splice(index, 1);
```

Reply:
Fixed.